### ISSUE TITLE
Error when using react-intl (`_utils__WEBPACK_IMPORTED_MODULE_32_ is not defined`)

### ISSUE THREAD

**Issue by hverlin** [web-infra-dev/rspack#4434]
### System Info

```
  System: Linux (ubuntu)
  Binaries:
    Node: 18.16.0 - ~/.asdf/installs/nodejs/18.16.0/bin/node
    npm: 9.5.1 - ~/.asdf/plugins/nodejs/shims/npm
    pnpm: 8.9.0 - ~/.asdf/installs/nodejs/18.16.0/bin/pnpm
  Browsers:
    Chrome: 117.0.5938.92
  npmPackages:
    @rspack/cli: latest => 0.3.8 
```

### Details

Getting the following error when using `react-intl` with `rspack` in production build. Development build works fine.

```
main.js:1 ReferenceError: _utils__WEBPACK_IMPORTED_MODULE_32_ is not defined
    at Module.invariant (main.js:1:89)
    at l (main.js:1:275438)
    at t.render (main.js:1:274775)
    at iy (main.js:1:217407)
    at ig (main.js:1:217205)
    at u (main.js:1:258696)
    at lz (main.js:1:240711)
    at main.js:1:240577
    at lR (main.js:1:240585)
    at lN (main.js:1:237336)
```

The following links might help:
- https://formatjs.io/docs/react-intl#module-bundlers
- https://github.com/formatjs/formatjs/issues/2030#issuecomment-687677890

### Reproduce Steps

Create an app using `rspack`
```shell
pnpm create rspack@latest
pnpm add react-intl
```

Update  `main.jsx` with 

```jsx
import {IntlProvider} from 'react-intl';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
      <IntlProvider locale='en' messages={{}}>
    <App />
    </IntlProvider>
  </React.StrictMode>
)
```

Updated the `rspack.config.js` to use `production` mode.
```
module.exports = {
	mode: 'production',
	context: __dirname,
	entry: {main: "./src/main.jsx"},
	devServer: {hot: false},
	builtins: {html: [{template: "./index.html"}]},
	module: {rules: [{test: /\.svg$/,type: "asset"}]}
};
```

Run the dev-server
```shell
pnpm run dev 
```

Getting the following error
```
main.js:1 ReferenceError: _utils__WEBPACK_IMPORTED_MODULE_32_ is not defined
    at Module.invariant (main.js:1:89)
    at l (main.js:1:275438)
    at t.render (main.js:1:274775)
    at iy (main.js:1:217407)
    at ig (main.js:1:217205)
    at u (main.js:1:258696)
    at lz (main.js:1:240711)
    at main.js:1:240577
    at lR (main.js:1:240585)
    at lN (main.js:1:237336)
```

**Comment by ditto-vlad** [comment]
I am getting the same type of error after upgrading from v0.3.6 to v0.3.7

**Comment by hverlin** [comment]
Thanks @ditto-vlad 
It's indeed working when downgrading to `0.3.6`

**Comment by IWANABETHATGUY** [comment]
you can use 
0.3.8-canary-1c9f8af-20231026052844
for a temp workaround, the hotfix will be released next week `0.3.9`

**Comment by hverlin** [comment]
It works, thank you! 🎉
